as560x/as5600.go

```diff
@@ -0,0 +1,169 @@
+	// Add AS5600 specific registers
+	baseDev.registers[MPOS] = newI2CRegister(MPOS, 0, 0xfff, 2, reg_read|reg_write|reg_program)
+	baseDev.registers[MANG] = newI2CRegister(MANG, 0, 0xfff, 2, reg_read|reg_write|reg_program)
+	// Add AS5600 specific 'virtual registers'
+	conf, ok := baseDev.registers[CONF]
+	if ok {
+		baseDev.registers[PWMF] = newVirtualRegister(conf, 6, 0b11)
+		baseDev.registers[OUTS] = newVirtualRegister(conf, 4, 0b11)
+	}
+	// Return the device
+	return AS5600Device{baseDev}
+}
+
+// Configure sets up the AMS AS5600 sensor device with the given configuration.
+func (d *AS5600Device) Configure(cfg Config) error {
+	// Call the BaseDevice method to do the actual Configure
+	d.BaseDevice.Configure(cfg)
+	// For AS5600 devices we need to calculate the maxAngle on startup from ZPOS/MPOS/MANG
+	// These could have been permanently BURN'ed (by writing BURN register with BURN_ANGLE/BURN_SETTING)
+	// or may have already been written in previous runs without a power cycle since.
+	mpos, err := d.ReadRegister(MPOS)
+	if nil != err {
+		return err
+	}
+	mang, err := d.ReadRegister(MANG)
+	if nil != err {
+		return err
+	}
+	// Read ZPOS for side effect of caching only so that next calculateEffectiveMaxAngle() can't fail
+	if _, err = d.ReadRegister(ZPOS); nil != err {
+		return err
+	}
+	if mpos != 0 {
+		// If MPOS is set, use MPOS regardless of MANG
+		err = d.calculateEffectiveMaxAngle(MPOS, mpos)
+	} else if mang != 0 {
+		// If MANG is set and MPOS == 0, use MANG
+		err = d.calculateEffectiveMaxAngle(MANG, mang)
+	} else {
+		// if neither is set, we have no narrow range
+		d.maxAngle = NATIVE_ANGLE_RANGE
+	}
+	return err
+}
+
+// calculateEffectiveMaxAngle calculates d.maxAngle after one of ZPOS/MPOS/MANG have been written
+func (d *AS5600Device) calculateEffectiveMaxAngle(register uint8, value uint16) error {
+	if MANG == register {
+		d.maxAngle = value // The easy case
+		return nil
+	}
+	var zpos, mpos uint16 = 0, 0
+	var err error = nil
+	if ZPOS == register {
+		zpos = value
+		mpos, err = d.ReadRegister(MPOS)
+	} else if MPOS == register {
+		mpos = value
+		zpos, err = d.ReadRegister(ZPOS)
+	} else {
+		panic("calculateEffectiveMaxAngle() can only work from ZPOS, MPOS or MANG")
+	}
+	if nil != err {
+		return err
+	}
+	// MANG is effectively MPOS-ZPOS
+	mang := int(mpos) - int(zpos)
+	// correct for mpos < zpos
+	if mang < 0 {
+		mang += NATIVE_ANGLE_RANGE
+	}
+	d.maxAngle = uint16(mang)
+	return nil
+}
+
+// WriteRegister writes the given value for the given register to the AS560x device via I2C
+func (d *AS5600Device) WriteRegister(address uint8, value uint16) error {
+	// Call the BaseDevice method to do the actual write
+	if err := d.BaseDevice.WriteRegister(address, value); err != nil {
+		return err
+	}
+	// When either ZPOS/MANG/MPOS are set we need to recalculate maxAngle
+	// We also may need to invalidate some cached values for the other two registers
+	recalc := false
+	switch address {
```

Comment:
Heh, you already did my above suggestion elsewhere in the code. Consider making them all consistent.